packages/dds/tree/src/shared-tree/treeApi.ts

```diff
@@ -470,13 +478,25 @@ function runTransactionInCheckout<TResult>(
 						`Attempted to add a "nodeInDocument" constraint, but the node is not currently in the document. Node status: ${nodeStatus}`,
 					);
 				}
-				checkout.editor.addNodeExistsConstraint(node.anchorNode);
+				checkout.editor.addInputNodeExistsConstraint(node.anchorNode);
 				break;
 			}
 			default:
 				unreachableCase(constraint.type);
 		}
 	}
+	for (const constraint of undoPreconditions) {
+		switch (constraint.type) {
+			case "nodeInDocument": {
+				const node = getOrCreateInnerNode(constraint.node);
+				checkout.editor.addUndoNodeExistsConstraint(node.anchorNode);
```

Comment:
Reminder: once we switch to the new API, we'll want to add these constraints after the `transaction()` call.

Reply:
Yes, I removed this from here. Will do it in a separate PR.